store/mongo/datastore_mongo.go

```diff
@@ -306,6 +307,76 @@ func (db *DataStoreMongo) CreateIntegration(
 	return &integration, err
 }
 
+func (db *DataStoreMongo) SetIntegrationCredentials(
+	ctx context.Context,
+	integrationId uuid.UUID,
+	credentials model.Credentials,
+) error {
+	collIntegrations := db.client.Database(DbName).Collection(CollNameIntegrations)
+
+	fltr := bson.D{{
+		Key:   KeyID,
+		Value: integrationId,
+	}}
+
+	update := bson.M{
+		"$set": bson.D{
+			{
+				Key:   KeyCredentials,
+				Value: credentials,
+			},
+		},
+	}
+
+	_, err := collIntegrations.UpdateOne(ctx,
+		mstore.WithTenantID(ctx, fltr),
+		update,
+		mopts.Update(),
+	)
+
+	return errors.Wrap(err, "mongo: failed to set integration credentials")
+}
+
+func (db *DataStoreMongo) RemoveIntegration(ctx context.Context, integrationId uuid.UUID) error {
+	collIntegrations := db.client.Database(DbName).Collection(CollNameIntegrations)
+	fltr := bson.D{{
+		Key:   KeyID,
+		Value: integrationId,
+	}}
+	_, err := collIntegrations.DeleteOne(ctx, mstore.WithTenantID(ctx, fltr))
+	return err
+}
+
+// GetDevicesByIntegrationID searches for all devices connected with given integration ID
+func (db *DataStoreMongo) GetDevicesByIntegrationID(
+	ctx context.Context,
+	integrationID uuid.UUID,
+) ([]model.Device, error) {
+	var (
+		err     error
+		results = []model.Device{}
+	)
+	if _, err := db.GetIntegrationById(ctx, integrationID); err != nil {
+		return nil, errors.Wrap(err, "error retrieving integration collection results")
+	}
+
+	collDevices := db.client.Database(*db.DbName).Collection(CollNameDevices)
+	fltr := bson.D{
+		{
+			Key: KeyIntegrationIDs, Value: integrationID,
+		},
+	}
+
+	cursor, err := collDevices.Find(ctx, mstore.WithTenantID(ctx, fltr))
+	if err != nil {
+		return nil, errors.Wrap(err, "error executing devices collection request")
+	}
+	if err = cursor.All(ctx, &results); err != nil {
+		return nil, errors.Wrap(err, "error retrieving devices collection results")
+	}
+	return results, err
```

Comment:
This query can be very expensive. Please replace it with a `Count`, returning the number of devices (which is what you want to test anyway). I suggest renaming this method to `GetDevicesCountByIntegrationID`.

Reply:
Alternatively, expose the `limit` (and/or `skip`) parameter and limit the result to a single result since we're only interested in checking if such a device exists in this case. I'm not sure how the count operator works internally, but if the same query needs to be executed it is just as expensive - unless the index data structure keeps track of sub-counts.

Reply:
I used limit and renamed the method, thanks!